src/psm/src/ir_solver.cpp

```diff
@@ -419,20 +419,87 @@ bool IRSolver::CreateJ() {  // take current_map as an input?
       continue;
     }
     inst->getLocation(x, y);
-    int l = m_bottom_layer;  // atach to the bottom most routing layer
-    Node* node_J = m_Gmat->GetNode(x, y, l, true);
-    NodeLoc node_loc = node_J->GetLoc();
-    if (abs(node_loc.first - x) > m_node_density ||
-        abs(node_loc.second - y) > m_node_density) {
-      m_logger->warn(utl::PSM, 24,
-                     "Instance {}, current node at ({}, {}) at layer {} have "
-                     "been moved from ({}, {}).",
-                     it->first, node_loc.first, node_loc.second, l, x, y);
+    int l = m_bottom_layer;  // attach to the bottom most routing layer
+    // Special condition to distribute power across multiple nodes for macro
+    // blocks
+    if(inst->isBlock() || inst->isPad()) {
+      dbBox* inst_bBox = inst->getBBox();
+      std::set<int> pin_layers;
+      auto iterms = inst->getITerms();
+      // Find the pin layers for the macro
+      for (auto&& iterm : iterms) {
+        if (iterm->getSigType() == m_power_net_type) {
+          auto mterm = iterm->getMTerm();  
+          for (auto mpin : mterm->getMPins()) {
+            for (auto box : mpin->getGeometry()) {
+              dbTechLayer* pin_layer = box->getTechLayer();
+              pin_layers.insert(pin_layer->getRoutingLevel());
+            }
+          }
+        }
+      }
+      // Search for all nodes within the macro boundary 
+      vector <Node*> nodes_J;
+      for(auto ll : pin_layers){
+        vector <Node*> nodes_J_l = m_Gmat->GetNodes(ll, inst_bBox->xMin(),
+                                                        inst_bBox->xMax(), 
+                                                        inst_bBox->yMin(), 
+                                                        inst_bBox->yMax());
+        nodes_J.insert(nodes_J.end(), nodes_J_l.begin(), nodes_J_l.end());
+      }
+      double num_nodes = nodes_J.size();
+      // If nodes are not found on the pin layers we search one layer above  
+      if(num_nodes == 0){
+        int max_l = *std::max_element(pin_layers.begin(),pin_layers.end());
+        for(int pl = m_bottom_layer+1; pl <= m_top_layer; pl++) {
```

Comment:
Is m_bottom_layer+1 " one layer above" - it sounds like the second metal layer?

Reply:
That is correct. the comment is outdated as that was the case previously, it now implemented as your question. I have modified the comment to reflect this.